### PR TITLE
Check that `auto_vacuum` is disabled when porting a SQLite database to Postgres, as `VACUUM`s must not be performed between runs of the script.

### PR DESCRIPTION
**Pull request by reivilibre**
The portdb script relies on `rowid`s being consistent.

`VACUUM` does not uphold that.

`auto_vacuum` is not enabled by default (unless SQLite was compiled with an option to make it default), but I'm aware that
some community members manually VACUUM and I wouldn't be surprised if some of them enabled `auto_vacuum` themselves.

Doing so is dangerous for the portdb script as it means that some rows may be omitted from the copy.

For safety's sake, let's prohibit `auto_vacuum` when porting a database **and** document that manual `VACUUM` are also not to be done.

### CODE REVIEW

File: synapse/_scripts/synapse_port_db.py [hunk 1]
```diff
@@ -637,6 +656,21 @@ async def run(self) -> None:
...
+                "is_sqlite_autovacuum_enabled", self._is_sqlite_autovacuum_enabled
+            ):
+                end_error = (
+                    "auto_vacuum is enabled in the SQLite database."
+                    " (This is not the default configuration.)\n"
+                    " This script relies on rowids being consistent and must not"
+                    " be used if the database could be vacuumed between re-runs.\n"
+                    " To disable auto_vacuum, you need to stop Synapse and run the following SQL:\n"
+                    " PRAGMA auto_vacuum=off;\n"
+                    " VACUUM;"
```
Comment (DMRobertson) [inline synapse/_scripts/synapse_port_db.py:670]: Why do you need to vacuum here after disabling autovacuum? To ensure there is no vacuum in progress when the script runs?

**Review by DMRobertson — APPROVED** [review]
Happy with the changes, just a few questions.

> The portdb script relies on rowids being consistent.

Can you point me to the relevant source code here?

File: synapse/_scripts/synapse_port_db.py (hunk 1 above)
Comment (reivilibre) [reply synapse/_scripts/synapse_port_db.py:670]: You can't change the `auto_vacuum` setting without `VACUUM`ing if there are existing tables

**Comment by reivilibre** [comment]
> > The portdb script relies on rowids being consistent.
> 
> Can you point me to the relevant source code here?

An example line is
https://github.com/matrix-org/synapse/blob/b0366853cae1e08d30d44757d142f8670b7ec35c/synapse/_scripts/synapse_port_db.py#L421

but feel free to look all over the script — the tables that are being migrated are paginated by rowid.

Notable is the `port_from_sqlite3` table that gets created, which tracks the oldest and newest rowids that have been copied across.

File: synapse/_scripts/synapse_port_db.py (hunk 1 above)
Comment (DMRobertson) [reply synapse/_scripts/synapse_port_db.py:670]: > You can't change the `auto_vacuum` setting without `VACUUM`ing if there are existing tables

Does that mean the vacuum should happen first?

Comment (reivilibre) [reply synapse/_scripts/synapse_port_db.py:670]: No; you set the PRAGMA then VACUUM to make it take effect.

**Pull request merged**